### PR TITLE
MGMT-6455 KubeAPI Controllers Debuggability: Add Logger Fields 

### PR DESCRIPTION
**Pull request by nmagnezi**
This change adds logger fields to controllers to specify the
resource name and namespace that is being reconciled.

Additionally, in a case that context does not have a request-id,
we'll generate one to track it down both in controllers and the backend.

Lastly, I added a log indicating that a reconcile iteration has ended.

### CODE REVIEW

**Comment by YuviGold** [comment]
Could we change the notation to something else than spaces?
Instead of 
```
		logrus.Fields{
			"agent":                       agent.Name,
			"agent namespace":             agent.Namespace,
			"clusterDeployment":           clusterDeployment.Name,
			"clusterDeployment namespace": clusterDeployment.Namespace,
		})
```

Maybe `clusterDeployment_namespace` or `clusterDeployment-namespace`
cc @filanov

File: internal/controller/controllers/agent_controller.go [hunk 1]
```diff
@@ -76,6 +76,12 @@ func (r *AgentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 		r.Log.WithError(err).Errorf("Failed to get resource %s", req.NamespacedName)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+	r.Log = r.Log.WithFields(
```
Comment (filanov) [inline internal/controller/controllers/agent_controller.go:79]: Please don't override the logger, create a new one from this one.

**Comment by filanov** [comment]
> Could we change the notation to something else than spaces?
> Instead of
> 
> ```
> 		logrus.Fields{
> 			"agent":                       agent.Name,
> 			"agent namespace":             agent.Namespace,
> 			"clusterDeployment":           clusterDeployment.Name,
> 			"clusterDeployment namespace": clusterDeployment.Namespace,
> 		})
> ```
> 
> Maybe `clusterDeployment_namespace` or `clusterDeployment-namespace`
> cc @filanov

Usually we call the fields with a snakecase for example` request_id`

File: internal/controller/controllers/agentserviceconfig_controller.go [hunk 2]
```diff
@@ -92,7 +92,8 @@ type AgentServiceConfigReconciler struct {
 // +kubebuilder:rbac:groups=route.openshift.io,resources=routes,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="",resources=events,verbs=create;patch
 
-func (r *AgentServiceConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *AgentServiceConfigReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (ctrl.Result, error) {
```
Comment (nmagnezi) [inline internal/controller/controllers/agentserviceconfig_controller.go:95]: this controller is using a different logger, so I couldn't use the request-id helpers with it.

**Comment by nmagnezi** [comment]
PR temporarily rebased on https://github.com/openshift/assisted-service/pull/1606 so I can preemptively resolve the git conflicts and make relevant modifications (added logs).
Once https://github.com/openshift/assisted-service/pull/1606 gets merged, I'll rebased on master.

**Review by djzager — COMMENTED** [review]
I think it will be nice for us to all use the same logger. I'll save the :+1: for someone else but this looks good to me.

**Review by filanov — APPROVED** [review]

**Pull request merged**